package/src/components/MessageInput/FileUploadPreview.tsx

```diff
@@ -113,72 +117,74 @@ const FileUploadPreviewWithContext = <
     },
   } = useTheme();
   const renderItem = ({ index, item }: { index: number; item: FileUpload }) => (
-    <UploadProgressIndicator
-      action={() => {
-        uploadFile({ newFile: item });
-      }}
-      active={item.state !== FileState.UPLOADED && item.state !== FileState.FINISHED}
-      style={styles.overlay}
-      type={
-        item.state === FileState.UPLOADING
-          ? ProgressIndicatorTypes.IN_PROGRESS
-          : item.state === FileState.UPLOAD_FAILED
-          ? ProgressIndicatorTypes.RETRY
-          : undefined
-      }
-    >
-      <View
-        style={[
-          styles.fileContainer,
-          index === fileUploads.length - 1
-            ? {
-                marginBottom: 0,
-              }
-            : {},
-          {
-            borderColor: grey_whisper,
-            width: flatListWidth - 16,
-          },
-          fileContainer,
-        ]}
+    <>
+      <UploadProgressIndicator
+        action={() => {
+          uploadFile({ newFile: item });
+        }}
+        active={item.state !== FileState.UPLOADED && item.state !== FileState.FINISHED}
+        style={styles.overlay}
+        type={
+          item.state === FileState.UPLOADING
+            ? ProgressIndicatorTypes.IN_PROGRESS
+            : item.state === FileState.UPLOAD_FAILED
+            ? ProgressIndicatorTypes.RETRY
+            : undefined
+        }
```

Comment:
I'm not sure about what naming makes proper sense here, but I mean something along these lines. `shouldAllowRetry` should probably have a different name if you can think of something more descriptive, I think.
```suggestion
        type={() => {
          const shouldAllowRetry = item.state === FileState.UPLOAD_FAILED
            ? ProgressIndicatorTypes.RETRY
            : undefined;
          
          return item.state === FileState.UPLOADING
            ? ProgressIndicatorTypes.IN_PROGRESS
            : shouldAllowRetry;
        }}
```